Neos.Http.Factories/Classes/HttpFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Http\Factories;
+
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ServerRequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Message\UploadedFileFactoryInterface;
+use Psr\Http\Message\UriFactoryInterface;
+
+class HttpFactory implements ServerRequestFactoryInterface, RequestFactoryInterface, ResponseFactoryInterface, UriFactoryInterface, StreamFactoryInterface, UploadedFileFactoryInterface
```

Comment:
I think the name needs some more thinking. This factory doesn't "create HTTP"…

Reply:
Agree ... suggestions are welcome aswell as an idea how to deal with the __constructor of the ServerRequestFactory

Reply:
Actually telling someone that the way to do a single http request is to inject 3 factories and plus Browser/Client (at least the engine is now injected by default) is an interesting experience.

Reply:
yeha, I thought about that too, you can always sacrifice portability by directly instantiating a specific implementation that you then obvs need to require in your package.

Reply:
How is "supposed to work" according to "the spec"? Is there anything in the PSR or in Guzzle?

Reply:
> [](https://www.php-fig.org/psr/psr-17/#2-interfaces)2. Interfaces
The following interfaces MAY be implemented together within a single class or in separate classes.

https://www.php-fig.org/psr/psr-17/#2-interfaces

Reply:
The par looks to me weird but HttpObjectFactory sounds just fine

Reply:
Either seem fine to me :) Although HttpObjectFactory sounds a bit smoother

Reply:
Indeed. That factory doesn't produce "HTTP", after all. ;)

Reply:
@kitsunet @kdambekalns renamed to HttpObjectFactory